main.cpp

```diff
@@ -19,13 +19,44 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+		if (const auto *cSCExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast")) {
+			const auto& SM = *Result.SourceManager;
+			auto& FM = SM.getFileManager();
+			
+			const auto& LParenLoc = cSCExpr->getLParenLoc();
+			const auto& RParenLoc = cSCExpr->getRParenLoc();
+			
+			// change to static_cast
+			rewriter.RemoveText(LParenLoc, 1);
+			rewriter.InsertText(LParenLoc, "static_cast<", false, false);
+			rewriter.RemoveText(RParenLoc, 1);
+			rewriter.InsertText(RParenLoc, ">", false, false);
+			
+			// go to DeclRefExpr
+			const auto implicitFirst = cSCExpr->child_begin();
+			const auto implicitSecond = implicitFirst->child_begin();
+			const auto declExpIter = implicitSecond->child_begin();

```

Comment:
To retrieve `DeclRefExpr` you can use:
1. `getSubExprAsWritten()` - retrieves the cast subexpression as it was written in the source code, looking through any implicit casts or other intermediate nodes introduced by semantic analysis.
2. `getSubExpr()->IgnoreImpCasts()`:
  `getSubExpr()` retrieves the first cast subexpression.
  `ignoreImpCasts()` skips past any implicit casts which might surround this expression until
  reaching a fixed point. It skips the following AST nodes: `ImplicitCastExpr`, `FullExpr`.